### PR TITLE
fix: bugfix for compose draft

### PR DESCRIPTION
**Pull request by nuanyang233**
# Pull Request

## Related Issue

Closes FW-1215 FW-1218 FW-1213 FW-1216 FW-1214

## Checklist

-   [ ] I have tested these changes locally.
-   [ ] I have reviewed the code changes.
-   [ ] I have added unit tests if applicable.

### CODE REVIEW

File: src/components/Compose/ComposeSend.tsx [hunk 1]
```diff
@@ -57,9 +59,11 @@ export function ComposeSend(props: ComposeSendProps) {
                 });
             }
             await delay(300);
+            // If the draft is applied and sent successfully, remove the draft.
+            if (draftId) removeDraft(draftId);
```
Comment (guanbinrui) [inline src/components/Compose/ComposeSend.tsx:63]: Clear the compose modal here?

https://github.com/DimensionDev/firefly.mask.social/blob/a204a6e7788c23bf98cd2ca55540caac79de1dd1/src/modals/ComposeModal.tsx#L229

File: src/components/Compose/ComposeThreadContent.tsx [hunk 2]
```diff
@@ -30,7 +30,7 @@ export function ComposeThreadContent(props: ComposeThreadContentProps) {
     return (
         <div>
             {posts.map((x, i) => {
-                const isSuccessFully = !!compact(values(x.postId)).length;
+                const isSuccessFully = compact(values(x.postId)).length === x.availableSources.length;
```
Comment (guanbinrui) [inline src/components/Compose/ComposeThreadContent.tsx:33]:
```suggestion
                const isSucceed = compact(values(x.postId)).length === x.availableSources.length;
```

File: src/modals/ComposeModal.tsx [hunk 3]
```diff
@@ -235,10 +236,24 @@ export const ComposeModalUI = forwardRef<SingletonModalRefCreator<ComposeModalPr
...
+                        flatten(
+                            posts.map((x) => {
+                                // Failed source obtained
+                                return compact(
+                                    Object.entries(x.postError).map(([key, value]) => (value ? key : undefined)),
+                                );
+                            }),
+                        ),
+                    ),
+                ] as SocialSource[];
```
Comment (guanbinrui) [inline src/modals/ComposeModal.tsx:254]: Try to compute in the store.

File: src/modals/ComposeModal.tsx [hunk 4]
```diff
@@ -264,11 +279,11 @@ export const ComposeModalUI = forwardRef<SingletonModalRefCreator<ComposeModalPr
                         draftId: draftId || uuid(),
                         createdAt: new Date(),
                         cursor,
-                        posts,
+                        posts: hasError
```
Comment (guanbinrui) [inline src/modals/ComposeModal.tsx:282]: Try to compute in the store.

File: src/components/Compose/ComposeSend.tsx (hunk 1 above)
Comment (nuanyang233) [reply src/components/Compose/ComposeSend.tsx:63]: Closing the modal will automatically clear.

File: src/modals/ComposeModal.tsx (hunk 3 above)
Comment (nuanyang233) [reply src/modals/ComposeModal.tsx:254]: No, I believe this is just a calculation when saving drafts, there's no need to compute in the store.

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**